### DISCUSSION TITLE
📣 New project based on Cloud API instead of this library

### DISCUSSION THREAD

**Discussion by cyntler** [cyntler/react-doc-viewer#300] [Announcements]
Hello! 👋

I have nothing but an idea only in my head about what I want to implement as a **substitute** for this (still imperfect) library. I decided to describe it here. I think it's a bit of a waste of time to develop something that has technical limitations anyway and sooner or later we will meet problems here.

### What are these imperfections? ❌

1. A very large number of people using this library still ask about support for rendering **Microsoft's documents** in a browser. Unfortunately, it is now based on the official preview service served by Microsoft, based on **iframe**s and having considerable limitations. And the biggest drawback here is the need to use public URL for the file.

2. The initial library was created without taking care of consistency in rendering and without basic functions to manage the state of displaying the document, listening to events, etc. Implementing this now is actually rewriting everything from scratch.

3. The original version of the library was created quite a long time ago, its rewriting will now be time-consuming and not very optimal.

### What idea do I have? 💡

I want to build a service that provides REST API that process files of various formats through the back-end and a front-end library along with wrappers for popular frameworks that will be able to **disply these processed files**. The library would only be a visual layer + the ability to download the viewed file if the owner of the application that embeds the library wants such functionality. REST API would actually act as a **transformer** of various files to a visually friendly version and also handle MS Office files, etc. Audio/video files would also be supported. Apart from the documents. Documents would be converted into PDFs on the back-end side and presented in any form, selected (including steeling) by the user.

### My lack of time ⌛️

The last few months have been difficult for me to find time for “free” work on the library. If I have to work on something, especially on a library that supports rendering multiple file formats, I would like to get something out of it and create something like SaaS. I would like to provide a really good product.

### Questions for you ❓

1. Would anyone be interested in this type of service?

2. What problems do you see for this type of solution?

3. Do you know the currently working solutions of this type? If so, why do you use this library instead of this solution?

I invite you to discuss!

**Comment by Raza410** [comment]
hi @cyntler , I am interested to work with you on this, let's connect and have detailed conversation. I have already seen some paid libraries which work have advanced capabilities so we can study them as well.

**Comment by nomad-ajit** [comment]
@cyntler, We used mammoth js to display docx files (Not perfect). I would love to contribute more to this project.
